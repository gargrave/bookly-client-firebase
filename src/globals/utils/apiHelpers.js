// @flow
import type { FbDocRef } from '../../modules/core/flowtypes';

async function getDocRef(
  db: any,
  table: string,
  id: string,
): FbDocRef {
  return db.collection(table).doc(id);
}

export {
  getDocRef,
};
