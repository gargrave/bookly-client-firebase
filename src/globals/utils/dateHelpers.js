export const toUnixTimestamp = (date, defaultValue) => {
  if (!date) {
    if (defaultValue === null || defaultValue === undefined) {
      return Date.now();
    }
    return defaultValue;
  }

  if (date._isAMomentObject) {
    return date.unix();
  }

  if (typeof date === 'number') {
    return date;
  }

  return defaultValue;
};

export const toUnixMs = (date) =>
  (date && date * 1000) || 0;
