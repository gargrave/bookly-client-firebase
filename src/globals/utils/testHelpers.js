import React from 'react';
import { shallow } from 'enzyme';

export class ComponentBuilder {
  constructor(component, defaultProps) {
    this.component = component;
    this.defaultProps = defaultProps;
  }

  shallowGetComponent(extraProps = {}) {
    const props = {
      ...this.defaultProps,
      ...extraProps,
    };
    return shallow(<this.component {...props} />);
  }
}
