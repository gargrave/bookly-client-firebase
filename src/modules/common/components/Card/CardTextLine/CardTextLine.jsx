// @flow
import React from 'react'
import { oneOf, string } from 'prop-types'
import styled from 'react-emotion'

import { colors } from '../../../../../styles'

type Props = {
  text: string,
  type?: string,
}

const validTypes = ['header', 'subtext', 'text', 'title']

const Styled = styled('div')`
  color: ${colors.textLight};
  margin-bottom: 0;

  &.bold,
  &.title {
    color: ${colors.textMed};
    font-weight: bold;
  }

  &.subtext {
    color: ${colors.textMedLight};
    font-size: 0.9em;
  }
`

const CardTextLine = ({ text, type = 'text' }: Props) => {
  return <Styled className={type || null}>{text}</Styled>
}

CardTextLine.propTypes = {
  text: string.isRequired,
  type: oneOf(validTypes),
}

export default CardTextLine
