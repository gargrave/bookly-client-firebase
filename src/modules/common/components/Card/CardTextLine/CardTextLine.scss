$textColor: #6e6969; // TODO: move to _colors

.cardTextLine {
  color: $textColor;
  margin-bottom: 0; // Shoelace override
}

.bold {
  font-weight: bold;
}
