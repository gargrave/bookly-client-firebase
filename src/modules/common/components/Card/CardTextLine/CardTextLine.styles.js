import styled from 'react-emotion'

import { colors } from '../../../../../styles'

export const CardTextLine = styled('div')`
  color: ${colors.textLight};
  margin-bottom: 0;

  &.bold,
  &.title {
    color: ${colors.textMed};
    font-weight: bold;
  }

  &.subtext {
    color: ${colors.textMedLight};
    font-size: 0.9em;
  }
`
