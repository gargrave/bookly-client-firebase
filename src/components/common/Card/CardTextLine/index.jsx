// @flow
import React from 'react';
import { bool, string } from 'prop-types';

import { buildClasses } from '../../../../utils/cssHelpers';

type Props = {
  bold?: boolean,
  condensed?: boolean,
  text: string,
};

const CardTextLine = ({
  bold = false,
  condensed = false,
  text,
}: Props) => {
  return (
    <p
      className={buildClasses([
        'card__text',
        bold ? 'card__text--bold' : '',
        condensed ? 'condensed' : '',
      ])}
    >
      {text}
    </p>
  );
};

CardTextLine.propTypes = {
  bold: bool,
  condensed: bool,
  text: string.isRequired,
};

export default CardTextLine;
