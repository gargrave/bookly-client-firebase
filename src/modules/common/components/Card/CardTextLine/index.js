import CardTextLine from './CardTextLine'

export default CardTextLine
