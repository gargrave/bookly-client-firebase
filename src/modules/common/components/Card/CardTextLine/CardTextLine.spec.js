import { ComponentBuilder } from '../../../../../utils/testHelpers'

import CardTextLine from './CardTextLine'

const defaultProps = {
  text: 'this is the text',
  type: 'text',
}

const builder = new ComponentBuilder(CardTextLine, defaultProps)

describe('CardTextLine', () => {
  let component

  describe('basic rendering', () => {
    test('renders correct', () => {
      component = builder.shallowGetComponent()
      expect(component).toMatchSnapshot()
      expect(component.render().text()).toBe(defaultProps.text)
    })
  })

  describe('conditional classes', () => {
    test('correctly applies the "text" class', () => {
      component = builder.shallowGetComponent()
      expect(component.find('.text').length).toBe(1)
      expect(component.find('.subtext').length).toBe(0)
      expect(component.find('.title').length).toBe(0)
    })

    test('correctly applies the "subtext" class', () => {
      component = builder.shallowGetComponent({ type: 'subtext' })
      expect(component.find('.text').length).toBe(0)
      expect(component.find('.subtext').length).toBe(1)
      expect(component.find('.title').length).toBe(0)
    })

    test('correctly applies the "title" class', () => {
      component = builder.shallowGetComponent({ type: 'title' })
      expect(component.find('.text').length).toBe(0)
      expect(component.find('.subtext').length).toBe(0)
      expect(component.find('.title').length).toBe(1)
    })
  })
})
