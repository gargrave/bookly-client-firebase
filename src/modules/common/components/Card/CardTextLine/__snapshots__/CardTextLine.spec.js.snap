// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTextLine basic rendering renders correct 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardTextLine
    text="this is the text"
    type="text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "this is the text",
      "className": "text",
    },
    "ref": null,
    "rendered": "this is the text",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "this is the text",
        "className": "text",
      },
      "ref": null,
      "rendered": "this is the text",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
